src/stirling/utils/linux_headers.cc

```diff
@@ -498,31 +498,21 @@ Status LinkHostLinuxHeaders(const std::filesystem::path& lib_modules_dir) {
   return Status::OK();
 }
 
-Status ExtractPackagedHeaders(PackagedLinuxHeadersSpec* headers_package) {
-  std::string expected_directory =
-      absl::Substitute("/usr/src/linux-headers-$0.$1.$2-pl", headers_package->version.version,
-                       headers_package->version.major_rev, headers_package->version.minor_rev);
-
-  // This is a loose check that we don't clobber what we *think* should the output directory.
-  // If someone built a tar.gz with an incorrect directory structure, this check wouldn't save us.
-  if (fs::Exists(expected_directory)) {
-    return error::Internal("Refusing to clobber existing directory: $0.", expected_directory);
-  }
-
-  // Extract the files.
+Status ExtractPackagedHeaders(const PackagedLinuxHeadersSpec* headers_package,
```

Comment:
I think it is possible to remove the arg. `const PackagedLinuxHeadersSpec* headers_package` and just use the two other args. Should make these args. (though) either `const std::string&` or `std::string_view`.

Reply:
I think the minitar constructor needs to take the path to the tarball at `headers_package->path`, so we have to pass it into `ExtractPackagedHeaders` for this line to work properly.   
```
::px::tools::Minitar minitar(headers_package->path.string());
```
Does that make sense?

Reply:
Ah. Makes sense.

Reply:
We should make that arg a const ref though.

Reply:
should be:

`const PackagedLinuxHeadersSpec& headers_package`